src/meta/src/manager/streaming_job.rs

```diff
@@ -299,7 +299,7 @@ impl StreamingJob {
         }
     }
 
-    // TODO: record all objects instead.
+    // TODO: to be removed, pass all objects uniformly through `dependencies` field instead.
     pub fn dependent_relations(&self) -> Vec<u32> {
```

Comment:
It seems `table.dependent_relations` is not set any more. Will it break sth to call this function?

Reply:
Oh. The caller uses `extend`

Reply:
I would be in favor of removing this field together now to avoid mistakes. Can put in an upstack PR and merge them together.

Reply:
> It seems `table.dependent_relations` is not set any more. Will it break sth to call this function?

I merged the result of this method and the new `dependencies` field to avoid breaking. And now it basically return empty vector for `MaterializedView` and `Sink`, and remain the same for `Table` and `Index`.

Since this PR is mainly for `drop function` check, I only changed the behavior of `MaterializedView` and `Sink`. Added TODO comments for the remaining ones.